HidPkg/UefiHidDxeV2/src/keyboard.rs

```diff
@@ -416,12 +436,10 @@ impl KeyboardHidHandler {
 
     /// Called to send LED state to the device.
     pub fn send_led_reports(&mut self, hid_io: &dyn HidIo) -> Result<(), efi::Status> {
-        for (id, output_report) in self.generate_led_output_reports() {
-            let result = hid_io.set_output_report(id.map(|x| u32::from(x) as u8), &output_report);
-            if let Err(result) = result {
-                debugln!(DEBUG_ERROR, "unexpected error sending output report: {:?}", result);
-                return Err(result);
-            }
+        let output_reports = self.generate_led_output_reports();
+        if let Err(result) = self.send_output_reports(hid_io, output_reports) {
+            debugln!(DEBUG_ERROR, "unexpected error sending output report: {:?}", result);
+            return Err(result);
         }
         Ok(())

```

Comment:
I think the error handling in the block can be simplified a bit.


```suggestion
        let output_reports = self.generate_led_output_reports();
        self.send_output_reports(hid_io, output_reports).map_err(|err| {
            debugln!(DEBUG_ERROR, "unexpected error sending output report: {:?}", err);
            err
        })
```

Reply:
Even simpler.


```suggestion
        let output_reports = self.generate_led_output_reports();
        self.send_output_reports(hid_io, output_reports).inspect_err(|err| {
            debugln!(DEBUG_ERROR, "unexpected error sending output report: {:?}", err);
        })
```